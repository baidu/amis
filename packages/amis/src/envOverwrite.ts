/**
 * @file 用于在移动端或不同语言环境下使用不同配置
 */

import {SchemaNode, Schema} from './types';
import {RendererProps, RendererConfig, addSchemaFilter} from './factory';
import {findObjectsWithKey} from './utils/helper';
const isMobile = (window as any).matchMedia?.('(max-width: 768px)').matches
  ? true
  : false;

// 这里不能用 addSchemaFilter 是因为还需要更深层的替换，比如 select 里的 options
export const envOverwrite = (schema: any, locale?: string) => {
  if (schema.mobile && isMobile) {
    Object.assign(schema, schema.mobile);
    delete schema.mobile;
  }

  if (locale) {
    let schemaNodes = findObjectsWithKey(schema, locale);
    for (let schemaNode of schemaNodes) {
      Object.assign(schemaNode, schemaNode[locale]);
      delete schemaNode[locale];
    }
  }

  if (isMobile) {
    let schemaNodes = findObjectsWithKey(schema, 'mobile');
    for (let schemaNode of schemaNodes) {
      Object.assign(schemaNode, schemaNode['mobile']);
      delete schemaNode['mobile'];
    }
  }
};
