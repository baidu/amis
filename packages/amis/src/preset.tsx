import {
  addRootWrapper,
  extendDefaultEnv,
  LazyComponent,
  render,
  themeable,
  ThemeProps
} from 'amis-core';
import {ImageGallery} from 'amis-ui';
import {setRenderSchemaFn} from 'amis-ui/lib/components/Alert';
import {alert, confirm} from 'amis-ui/lib/components/Alert';
import {toast} from 'amis-ui/lib/components/Toast';
import React from 'react';

extendDefaultEnv({
  alert,
  confirm,
  notify: (type: keyof typeof toast, msg: string, conf: any) =>
    toast[type] ? toast[type](msg, conf) : console.warn('[Notify]', type, msg)
});

setRenderSchemaFn((controls, value, callback, scopeRef, theme) => {
  return render(
    {
      name: 'form',
      type: 'form',
      wrapWithPanel: false,
      mode: 'horizontal',
      controls,
      messages: {
        validateFailed: ''
      }
    },
    {
      data: value,
      onFinished: callback,
      scopeRef,
      theme
    },
    {
      session: 'prompt'
    }
  );
});

addRootWrapper((props: any) => {
  const {env, children} = props;
  return (
    <ImageGallery modalContainer={env.getModalContainer}>
      {children}
    </ImageGallery>
  );
});

const SimpleSpinner = themeable(
  (
    props: {
      className?: string;
      spinnerClassName?: string;
    } & ThemeProps
  ) => {
    const cx = props.classnames;

    return (
      <div
        data-testid="spinner"
        className={cx(`Spinner`, 'in', props.className)}
      >
        <div
          className={cx(
            `Spinner-icon`,
            'Spinner-icon--default',
            'Spinner-icon--sm',
            props.spinnerClassName
          )}
        ></div>
      </div>
    );
  }
);

(LazyComponent as any).defaultProps.placeholder = <SimpleSpinner />;
