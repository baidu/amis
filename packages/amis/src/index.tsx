/** @license amis v@version
 *
 * Copyright Baidu
 *
 * This source code is licensed under the Apache license found in the
 * LICENSE file in the root directory of this source tree.
 */
export * from 'amis-core';
export * from 'amis-ui';

import './preset';

// 注册渲染器
import './renderers/Action';
import './renderers/Alert';
import './renderers/App';
import './renderers/Avatar';
import './renderers/Remark';
import './renderers/ButtonGroup';
import './renderers/Form/ButtonToolbar';
import './renderers/Breadcrumb';
import './renderers/DropDownButton';
import './renderers/Calendar';
import './renderers/Collapse';
import './renderers/CollapseGroup';
import './renderers/Color';
import './renderers/CRUD';
import './renderers/CRUD2';
import './renderers/Pagination';
import './renderers/Cards';
import './renderers/Card';
import './renderers/Card2';
import './renderers/Custom';
import './renderers/Date';
import './renderers/Dialog';
import './renderers/Divider';
import './renderers/Each';
import './renderers/Flex';
import './renderers/Form/Control';
import './renderers/Form/Hidden';
import './renderers/Form/InputText';
import './renderers/Form/InputTag';
import './renderers/Form/InputNumber';
import './renderers/Form/Textarea';
import './renderers/Form/Checkboxes';
import './renderers/Form/Checkbox';
import './renderers/Form/InputCity';
import './renderers/Form/ChartRadios';
import './renderers/Form/InputRating';
import './renderers/Form/Switch';
import './renderers/Form/Radios';
import './renderers/Form/Radio';
import './renderers/Form/JSONSchema';
import './renderers/Form/JSONSchemaEditor';
import './renderers/Form/ListSelect';
import './renderers/Form/LocationPicker';
import './renderers/Form/Select';
import './renderers/Form/Static';
import './renderers/Form/InputDate';
import './renderers/Form/InputDateRange';
import './renderers/Form/InputFormula';
import './renderers/Form/InputRepeat';
import './renderers/Form/InputTree';
import './renderers/Form/TreeSelect';
import './renderers/Form/InputImage';
import './renderers/Form/InputFile';
import './renderers/Form/UUID';
import './renderers/Form/MatrixCheckboxes';
import './renderers/Form/InputMonthRange';
import './renderers/Form/InputQuarterRange';
import './renderers/Form/InputYearRange';
import './renderers/Form/InputRange';
import './renderers/Form/InputArray';
import './renderers/Form/Combo';
import './renderers/Form/ConditionBuilder';
import './renderers/Form/InputSubForm';
import './renderers/Form/InputExcel';
import './renderers/Form/InputRichText';
import './renderers/Form/Editor';
import './renderers/Form/DiffEditor';
import './renderers/Form/InputColor';
import './renderers/Form/ChainedSelect';
import './renderers/Form/NestedSelect';
import './renderers/Form/Transfer';
import './renderers/Form/TransferPicker';
import './renderers/Form/InputTable';
import './renderers/Form/Picker';
import './renderers/Form/IconPicker';
import './renderers/Form/IconSelect';
import './renderers/Form/Formula';
import './renderers/Form/FieldSet';
import './renderers/Form/TabsTransfer';
import './renderers/Form/TabsTransferPicker';
import './renderers/Form/Group';
import './renderers/Form/InputGroup';
import './renderers/Form/UserSelect';
import './renderers/Grid';
import './renderers/Grid2D';
import './renderers/HBox';
import './renderers/VBox';
import './renderers/Image';
import './renderers/Images';
import './renderers/List';
import './renderers/Log';
import './renderers/Operation';
import './renderers/Page';
import './renderers/PaginationWrapper';
import './renderers/Panel';
import './renderers/Plain';
import './renderers/Property';
import './renderers/Portlet';
import './renderers/Spinner';
import './renderers/Table/index';
import './renderers/Tabs';
import './renderers/Tpl';
import './renderers/Mapping';
import './renderers/Progress';
import './renderers/Status';
import './renderers/Json';
import './renderers/Link';
import './renderers/Wizard';
import './renderers/Chart';
import './renderers/Container';
import './renderers/SwitchContainer';
import './renderers/SearchBox';
import './renderers/Service';
import './renderers/SparkLine';
import './renderers/Video';
import './renderers/Audio';
import './renderers/Nav';
import './renderers/Number';
import './renderers/Tasks';
import './renderers/Drawer';
import './renderers/Wrapper';
import './renderers/IFrame';
import './renderers/BarCode';
import './renderers/QRCode';
import './renderers/Icon';
import './renderers/Carousel';
import './renderers/AnchorNav';
import './renderers/Steps';
import './renderers/Timeline';
import './renderers/Markdown';
import './renderers/TableView';
import './renderers/Code';
import './renderers/WebComponent';
import './renderers/GridNav';
import './renderers/TooltipWrapper';
import './renderers/Tag';
import './renderers/Table2/index';
import './renderers/Words';
import './renderers/Password';
import './renderers/DateRange';
import './renderers/MultilineText';
import './renderers/OfficeViewer';
import './renderers/AMIS';

import './compat';
import './schemaExtend';
import type {
  BaseSchema,
  FormSchema,
  SchemaApi,
  SchemaCollection,
  SchemaExpression,
  SchemaObject
} from './Schema';
import type {TableViewSchema, TrObject} from './renderers/TableView';
import type {ActionSchema, ButtonSchema} from './renderers/Action';
import type {CRUDCommonSchema} from './renderers/CRUD';
import type {CRUD2Schema} from './renderers/CRUD2';
import type {TabsSchema} from './renderers/Tabs';
import {availableLanguages as EditorAvailableLanguages} from './renderers/Form/Editor';
import type {Action} from './types';
export * from './renderers/Form/IconPickerIcons';
export * from './renderers/Form/IconSelectStore';

export {
  BaseSchema,
  SchemaCollection,
  FormSchema,
  SchemaApi,
  SchemaObject,
  TableViewSchema,
  TrObject,
  ActionSchema,
  CRUDCommonSchema,
  ButtonSchema,
  CRUD2Schema,
  TabsSchema,
  SchemaExpression,
  Action,
  EditorAvailableLanguages
};
