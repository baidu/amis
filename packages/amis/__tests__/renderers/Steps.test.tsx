/**
 * 组件名称：Steps 步骤条
 * 单测内容：
 1. 基础使用
 2. 自定义状态 status
 3. 标签位置 labelPlacement
 4. 点状步骤条 progressDot
 5. 竖向排列 mode vertical
 6. 上下文数据 name
 */

import React from 'react';
import {fireEvent, render, waitFor} from '@testing-library/react';
import '../../src';
import {render as amisRender} from '../../src';
import {makeEnv, wait} from '../helper';

test('Renderer:steps', () => {
  const schema = {
    type: 'steps',
    value: 1,
    steps: [
      {
        title: 'First',
        subTitle: 'this is subTitle',
        description: 'this is description',
        icon: 'close',
        className: 'className',
        value: 1
      },
      {
        title: 'Second'
      },
      {
        title: 'Last'
      }
    ]
  };
  const {container, rerender} = render(amisRender(schema, {}, makeEnv({})));
  expect(container).toMatchSnapshot();

  schema.steps[0].value = undefined;
  schema.steps[1].value = 1;
  rerender(amisRender(schema, {}, makeEnv({})));
  expect(container).toMatchSnapshot();
});

test('Renderer:steps status', async () => {
  const {container} = render(
    amisRender(
      {
        type: 'steps',
        className: 'className',
        status: {
          a: 'finish',
          b: 'error',
          c: 'wait'
        },
        steps: [
          {
            title: 'First',
            value: 'a'
          },
          {
            title: 'Second',
            subTitle: 'this is subTitle',
            description: 'this is description',
            value: 'b'
          },
          {
            title: 'Third',
            value: 'c'
          }
        ]
      },
      {},
      makeEnv()
    )
  );

  const steps = container.querySelectorAll('.cxd-Steps .cxd-StepsItem');
  expect(steps!.length).toBe(3);
  expect(steps[0]).toHaveClass('is-finish');
  expect(steps[1]).toHaveClass('is-error');
  expect(steps[2]).toHaveClass('is-wait');

  expect(container).toMatchSnapshot();
});

test('Renderer:steps labelPlacement', async () => {
  const {container} = render(
    amisRender(
      {
        type: 'steps',
        labelPlacement: 'vertical',
        className: 'className',
        steps: [
          {
            title: 'First',
            value: 'a'
          },
          {
            title: 'Second',
            subTitle: 'this is subTitle',
            description: 'this is description',
            value: 'b'
          },
          {
            title: 'Third',
            value: 'c'
          }
        ]
      },
      {},
      makeEnv()
    )
  );

  expect(container.querySelector('.cxd-Steps')).toHaveClass(
    'cxd-Steps--Placement-vertical'
  );
  expect(container).toMatchSnapshot();
});

test('Renderer:steps progressDot', async () => {
  const {container} = render(
    amisRender(
      {
        type: 'steps',
        className: 'className',
        progressDot: true,
        steps: [
          {
            title: 'First',
            value: 'a'
          },
          {
            title: 'Second',
            subTitle: 'this is subTitle',
            description: 'this is description',
            value: 'b'
          },
          {
            title: 'Third',
            value: 'c'
          }
        ]
      },
      {},
      makeEnv()
    )
  );
  expect(container.querySelector('.cxd-Steps')).toHaveClass(
    'cxd-Steps--ProgressDot'
  );
  expect(container).toMatchSnapshot();
});

test('Renderer:steps with vertical mode', async () => {
  const {container} = render(
    amisRender({
      type: 'steps',
      className: 'className',
      progressDot: true,
      mode: 'vertical',
      steps: [
        {
          title: 'First',
          value: 'a'
        },
        {
          title: 'Second',
          subTitle: 'this is subTitle',
          description: 'this is description',
          value: 'b'
        },
        {
          title: 'Third',
          value: 'c'
        }
      ]
    })
  );

  expect(container.querySelector('.cxd-Steps')).toHaveClass(
    'cxd-Steps--vertical'
  );
  expect(container).toMatchSnapshot();
});

test('Renderer:steps with name', async () => {
  const {container} = render(
    amisRender({
      type: 'page',
      data: {
        step: 0,
        status: 'error',
        secondTitle: 'Second Test',
        desc: 'very good idea',
        what: 'what is this'
      },
      body: [
        {
          type: 'input-number',
          name: 'step'
        },
        {
          type: 'steps',
          name: 'step',
          status: '${status}',
          steps: [
            {
              title: 'First',
              subTitle: 'this is subTitle',
              description: '${desc}'
            },
            {
              title: '${secondTitle}'
            },
            {
              title: 'Last',
              subTitle: '${what}'
            }
          ]
        }
      ]
    })
  );

  const input = container.querySelector('.cxd-Number input');
  const setNum = async (num: number = 0) => {
    fireEvent.change(input!, {
      target: {value: num}
    });
    await wait(300);
  };

  const steps = container.querySelectorAll('.cxd-Steps .cxd-StepsItem');
  expect(steps!.length).toBe(3);

  expect(steps[0]).toHaveClass('is-error');
  expect(
    steps[0].querySelector('.cxd-StepsItem-description')
  ).toHaveTextContent('very good idea');

  await setNum(1);
  expect(steps[0]).toHaveClass('is-finish');
  expect(steps[1]).toHaveClass('is-error');
  expect(steps[2]).toHaveClass('is-wait');

  expect(steps[1].querySelector('.cxd-StepsItem-title')).toHaveTextContent(
    'Second Test'
  );

  await setNum(2);
  expect(steps[0]).toHaveClass('is-finish');
  expect(steps[1]).toHaveClass('is-finish');
  expect(steps[2]).toHaveClass('is-error');

  expect(steps[2].querySelector('.cxd-StepsItem-subTitle')).toHaveTextContent(
    'what is this'
  );

  await setNum(3);

  expect(steps[0]).toHaveClass('is-finish');
  expect(steps[1]).toHaveClass('is-finish');
  expect(steps[2]).toHaveClass('is-finish');
});
