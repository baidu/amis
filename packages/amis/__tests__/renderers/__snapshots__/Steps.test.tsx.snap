// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderer:steps 1`] = `
<div>
  <ul
    class="cxd-Steps cxd-Steps--Placement- cxd-Steps-- cxd-Steps--horizontal"
  >
    <li
      class="cxd-StepsItem is-process className"
    >
      <div
        class="cxd-StepsItem-container"
      >
        <div
          class="cxd-StepsItem-containerTail"
        />
        <div
          class="cxd-StepsItem-containerIcon"
        >
          <span
            class="cxd-StepsItem-icon"
          >
            <icon-mock
              classname="icon icon-close"
              icon="close"
            />
          </span>
        </div>
        <div
          class="cxd-StepsItem-containerWrapper"
        >
          <div
            class="cxd-StepsItem-body"
          >
            <div
              class="cxd-StepsItem-title cxd-StepsItem-"
            >
              <span
                class="cxd-StepsItem-ellText"
                title="First"
              >
                First
              </span>
              <span
                class="cxd-StepsItem-subTitle cxd-StepsItem-ellText"
                title="this is subTitle"
              >
                this is subTitle
              </span>
            </div>
            <div
              class="cxd-StepsItem-description cxd-StepsItem-ellText"
              title="this is description"
            >
              <span>
                this is description
              </span>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li
      class="cxd-StepsItem is-wait"
    >
      <div
        class="cxd-StepsItem-container"
      >
        <div
          class="cxd-StepsItem-containerTail"
        />
        <div
          class="cxd-StepsItem-containerIcon"
        >
          <span
            class="cxd-StepsItem-icon"
          >
            2
          </span>
        </div>
        <div
          class="cxd-StepsItem-containerWrapper"
        >
          <div
            class="cxd-StepsItem-body"
          >
            <div
              class="cxd-StepsItem-title cxd-StepsItem-"
            >
              <span
                class="cxd-StepsItem-ellText"
                title="Second"
              >
                Second
              </span>
            </div>
            <div
              class="cxd-StepsItem-description cxd-StepsItem-ellText"
            >
              <span />
            </div>
          </div>
        </div>
      </div>
    </li>
    <li
      class="cxd-StepsItem is-wait"
    >
      <div
        class="cxd-StepsItem-container"
      >
        <div
          class="cxd-StepsItem-containerTail"
        />
        <div
          class="cxd-StepsItem-containerIcon"
        >
          <span
            class="cxd-StepsItem-icon"
          >
            3
          </span>
        </div>
        <div
          class="cxd-StepsItem-containerWrapper"
        >
          <div
            class="cxd-StepsItem-body"
          >
            <div
              class="cxd-StepsItem-title cxd-StepsItem-"
            >
              <span
                class="cxd-StepsItem-ellText"
                title="Last"
              >
                Last
              </span>
            </div>
            <div
              class="cxd-StepsItem-description cxd-StepsItem-ellText"
            >
              <span />
            </div>
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;

exports[`Renderer:steps 2`] = `
<div>
  <ul
    class="cxd-Steps cxd-Steps--Placement- cxd-Steps-- cxd-Steps--horizontal"
  >
    <li
      class="cxd-StepsItem is-process className"
    >
      <div
        class="cxd-StepsItem-container"
      >
        <div
          class="cxd-StepsItem-containerTail"
        />
        <div
          class="cxd-StepsItem-containerIcon"
        >
          <span
            class="cxd-StepsItem-icon"
          >
            <icon-mock
              classname="icon icon-close"
              icon="close"
            />
          </span>
        </div>
        <div
          class="cxd-StepsItem-containerWrapper"
        >
          <div
            class="cxd-StepsItem-body"
          >
            <div
              class="cxd-StepsItem-title cxd-StepsItem-"
            >
              <span
                class="cxd-StepsItem-ellText"
                title="First"
              >
                First
              </span>
              <span
                class="cxd-StepsItem-subTitle cxd-StepsItem-ellText"
                title="this is subTitle"
              >
                this is subTitle
              </span>
            </div>
            <div
              class="cxd-StepsItem-description cxd-StepsItem-ellText"
              title="this is description"
            >
              <span>
                this is description
              </span>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li
      class="cxd-StepsItem is-wait"
    >
      <div
        class="cxd-StepsItem-container"
      >
        <div
          class="cxd-StepsItem-containerTail"
        />
        <div
          class="cxd-StepsItem-containerIcon"
        >
          <span
            class="cxd-StepsItem-icon"
          >
            2
          </span>
        </div>
        <div
          class="cxd-StepsItem-containerWrapper"
        >
          <div
            class="cxd-StepsItem-body"
          >
            <div
              class="cxd-StepsItem-title cxd-StepsItem-"
            >
              <span
                class="cxd-StepsItem-ellText"
                title="Second"
              >
                Second
              </span>
            </div>
            <div
              class="cxd-StepsItem-description cxd-StepsItem-ellText"
            >
              <span />
            </div>
          </div>
        </div>
      </div>
    </li>
    <li
      class="cxd-StepsItem is-wait"
    >
      <div
        class="cxd-StepsItem-container"
      >
        <div
          class="cxd-StepsItem-containerTail"
        />
        <div
          class="cxd-StepsItem-containerIcon"
        >
          <span
            class="cxd-StepsItem-icon"
          >
            3
          </span>
        </div>
        <div
          class="cxd-StepsItem-containerWrapper"
        >
          <div
            class="cxd-StepsItem-body"
          >
            <div
              class="cxd-StepsItem-title cxd-StepsItem-"
            >
              <span
                class="cxd-StepsItem-ellText"
                title="Last"
              >
                Last
              </span>
            </div>
            <div
              class="cxd-StepsItem-description cxd-StepsItem-ellText"
            >
              <span />
            </div>
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;

exports[`Renderer:steps labelPlacement 1`] = `
<div>
  <ul
    class="cxd-Steps cxd-Steps--Placement-vertical cxd-Steps-- cxd-Steps--horizontal className"
  >
    <li
      class="cxd-StepsItem is-process"
    >
      <div
        class="cxd-StepsItem-container"
      >
        <div
          class="cxd-StepsItem-containerTail"
        />
        <div
          class="cxd-StepsItem-containerIcon"
        >
          <span
            class="cxd-StepsItem-icon"
          >
            1
          </span>
        </div>
        <div
          class="cxd-StepsItem-containerWrapper"
        >
          <div
            class="cxd-StepsItem-body"
          >
            <div
              class="cxd-StepsItem-title cxd-StepsItem-"
            >
              <span
                class="cxd-StepsItem-ellText"
                title="First"
              >
                First
              </span>
            </div>
            <div
              class="cxd-StepsItem-description cxd-StepsItem-ellText"
            >
              <span />
            </div>
          </div>
        </div>
      </div>
    </li>
    <li
      class="cxd-StepsItem is-wait"
    >
      <div
        class="cxd-StepsItem-container"
      >
        <div
          class="cxd-StepsItem-containerTail"
        />
        <div
          class="cxd-StepsItem-containerIcon"
        >
          <span
            class="cxd-StepsItem-icon"
          >
            2
          </span>
        </div>
        <div
          class="cxd-StepsItem-containerWrapper"
        >
          <div
            class="cxd-StepsItem-body"
          >
            <div
              class="cxd-StepsItem-title cxd-StepsItem-"
            >
              <span
                class="cxd-StepsItem-ellText"
                title="Second"
              >
                Second
              </span>
              <span
                class="cxd-StepsItem-subTitle cxd-StepsItem-ellText"
                title="this is subTitle"
              >
                this is subTitle
              </span>
            </div>
            <div
              class="cxd-StepsItem-description cxd-StepsItem-ellText"
              title="this is description"
            >
              <span>
                this is description
              </span>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li
      class="cxd-StepsItem is-wait"
    >
      <div
        class="cxd-StepsItem-container"
      >
        <div
          class="cxd-StepsItem-containerTail"
        />
        <div
          class="cxd-StepsItem-containerIcon"
        >
          <span
            class="cxd-StepsItem-icon"
          >
            3
          </span>
        </div>
        <div
          class="cxd-StepsItem-containerWrapper"
        >
          <div
            class="cxd-StepsItem-body"
          >
            <div
              class="cxd-StepsItem-title cxd-StepsItem-"
            >
              <span
                class="cxd-StepsItem-ellText"
                title="Third"
              >
                Third
              </span>
            </div>
            <div
              class="cxd-StepsItem-description cxd-StepsItem-ellText"
            >
              <span />
            </div>
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;

exports[`Renderer:steps progressDot 1`] = `
<div>
  <ul
    class="cxd-Steps cxd-Steps--Placement-vertical cxd-Steps--ProgressDot cxd-Steps--horizontal className"
  >
    <li
      class="cxd-StepsItem is-process cxd-StepsItem-ProgressDot"
    >
      <div
        class="cxd-StepsItem-container"
      >
        <div
          class="cxd-StepsItem-containerTail"
        />
        <div
          class="cxd-StepsItem-containerProgressDot"
        />
        <div
          class="cxd-StepsItem-containerWrapper"
        >
          <div
            class="cxd-StepsItem-body"
          >
            <div
              class="cxd-StepsItem-title cxd-StepsItem-vertical-ProgressDot"
            >
              <span
                class="cxd-StepsItem-ellText"
                title="First"
              >
                First
              </span>
            </div>
            <div
              class="cxd-StepsItem-description cxd-StepsItem-ellText"
            >
              <span />
            </div>
          </div>
        </div>
      </div>
    </li>
    <li
      class="cxd-StepsItem is-wait cxd-StepsItem-ProgressDot"
    >
      <div
        class="cxd-StepsItem-container"
      >
        <div
          class="cxd-StepsItem-containerTail"
        />
        <div
          class="cxd-StepsItem-containerProgressDot"
        />
        <div
          class="cxd-StepsItem-containerWrapper"
        >
          <div
            class="cxd-StepsItem-body"
          >
            <div
              class="cxd-StepsItem-title cxd-StepsItem-vertical-ProgressDot"
            >
              <span
                class="cxd-StepsItem-ellText"
                title="Second"
              >
                Second
              </span>
              <span
                class="cxd-StepsItem-subTitle cxd-StepsItem-ellText"
                title="this is subTitle"
              >
                this is subTitle
              </span>
            </div>
            <div
              class="cxd-StepsItem-description cxd-StepsItem-ellText"
              title="this is description"
            >
              <span>
                this is description
              </span>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li
      class="cxd-StepsItem is-wait cxd-StepsItem-ProgressDot"
    >
      <div
        class="cxd-StepsItem-container"
      >
        <div
          class="cxd-StepsItem-containerTail"
        />
        <div
          class="cxd-StepsItem-containerProgressDot"
        />
        <div
          class="cxd-StepsItem-containerWrapper"
        >
          <div
            class="cxd-StepsItem-body"
          >
            <div
              class="cxd-StepsItem-title cxd-StepsItem-vertical-ProgressDot"
            >
              <span
                class="cxd-StepsItem-ellText"
                title="Third"
              >
                Third
              </span>
            </div>
            <div
              class="cxd-StepsItem-description cxd-StepsItem-ellText"
            >
              <span />
            </div>
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;

exports[`Renderer:steps status 1`] = `
<div>
  <ul
    class="cxd-Steps cxd-Steps--Placement- cxd-Steps-- cxd-Steps--horizontal className"
  >
    <li
      class="cxd-StepsItem is-finish"
    >
      <div
        class="cxd-StepsItem-container"
      >
        <div
          class="cxd-StepsItem-containerTail"
        />
        <div
          class="cxd-StepsItem-containerIcon"
        >
          <span
            class="cxd-StepsItem-icon"
          >
            1
          </span>
        </div>
        <div
          class="cxd-StepsItem-containerWrapper"
        >
          <div
            class="cxd-StepsItem-body"
          >
            <div
              class="cxd-StepsItem-title cxd-StepsItem-"
            >
              <span
                class="cxd-StepsItem-ellText"
                title="First"
              >
                First
              </span>
            </div>
            <div
              class="cxd-StepsItem-description cxd-StepsItem-ellText"
            >
              <span />
            </div>
          </div>
        </div>
      </div>
    </li>
    <li
      class="cxd-StepsItem is-error"
    >
      <div
        class="cxd-StepsItem-container"
      >
        <div
          class="cxd-StepsItem-containerTail"
        />
        <div
          class="cxd-StepsItem-containerIcon"
        >
          <span
            class="cxd-StepsItem-icon"
          >
            2
          </span>
        </div>
        <div
          class="cxd-StepsItem-containerWrapper"
        >
          <div
            class="cxd-StepsItem-body"
          >
            <div
              class="cxd-StepsItem-title cxd-StepsItem-"
            >
              <span
                class="cxd-StepsItem-ellText"
                title="Second"
              >
                Second
              </span>
              <span
                class="cxd-StepsItem-subTitle cxd-StepsItem-ellText"
                title="this is subTitle"
              >
                this is subTitle
              </span>
            </div>
            <div
              class="cxd-StepsItem-description cxd-StepsItem-ellText"
              title="this is description"
            >
              <span>
                this is description
              </span>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li
      class="cxd-StepsItem is-wait"
    >
      <div
        class="cxd-StepsItem-container"
      >
        <div
          class="cxd-StepsItem-containerTail"
        />
        <div
          class="cxd-StepsItem-containerIcon"
        >
          <span
            class="cxd-StepsItem-icon"
          >
            3
          </span>
        </div>
        <div
          class="cxd-StepsItem-containerWrapper"
        >
          <div
            class="cxd-StepsItem-body"
          >
            <div
              class="cxd-StepsItem-title cxd-StepsItem-"
            >
              <span
                class="cxd-StepsItem-ellText"
                title="Third"
              >
                Third
              </span>
            </div>
            <div
              class="cxd-StepsItem-description cxd-StepsItem-ellText"
            >
              <span />
            </div>
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;

exports[`Renderer:steps with vertical mode 1`] = `
<div>
  <ul
    class="cxd-Steps cxd-Steps--Placement-vertical cxd-Steps--ProgressDot cxd-Steps--vertical className"
  >
    <li
      class="cxd-StepsItem is-process cxd-StepsItem-ProgressDot"
    >
      <div
        class="cxd-StepsItem-container"
      >
        <div
          class="cxd-StepsItem-containerTail"
        />
        <div
          class="cxd-StepsItem-containerProgressDot"
        />
        <div
          class="cxd-StepsItem-containerWrapper"
        >
          <div
            class="cxd-StepsItem-body"
          >
            <div
              class="cxd-StepsItem-title cxd-StepsItem-vertical-ProgressDot"
            >
              <span
                class="cxd-StepsItem-ellText"
                title="First"
              >
                First
              </span>
            </div>
            <div
              class="cxd-StepsItem-description cxd-StepsItem-ellText"
            >
              <span />
            </div>
          </div>
        </div>
      </div>
    </li>
    <li
      class="cxd-StepsItem is-wait cxd-StepsItem-ProgressDot"
    >
      <div
        class="cxd-StepsItem-container"
      >
        <div
          class="cxd-StepsItem-containerTail"
        />
        <div
          class="cxd-StepsItem-containerProgressDot"
        />
        <div
          class="cxd-StepsItem-containerWrapper"
        >
          <div
            class="cxd-StepsItem-body"
          >
            <div
              class="cxd-StepsItem-title cxd-StepsItem-vertical-ProgressDot"
            >
              <span
                class="cxd-StepsItem-ellText"
                title="Second"
              >
                Second
              </span>
              <span
                class="cxd-StepsItem-subTitle cxd-StepsItem-ellText"
                title="this is subTitle"
              >
                this is subTitle
              </span>
            </div>
            <div
              class="cxd-StepsItem-description cxd-StepsItem-ellText"
              title="this is description"
            >
              <span>
                this is description
              </span>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li
      class="cxd-StepsItem is-wait cxd-StepsItem-ProgressDot"
    >
      <div
        class="cxd-StepsItem-container"
      >
        <div
          class="cxd-StepsItem-containerTail"
        />
        <div
          class="cxd-StepsItem-containerProgressDot"
        />
        <div
          class="cxd-StepsItem-containerWrapper"
        >
          <div
            class="cxd-StepsItem-body"
          >
            <div
              class="cxd-StepsItem-title cxd-StepsItem-vertical-ProgressDot"
            >
              <span
                class="cxd-StepsItem-ellText"
                title="Third"
              >
                Third
              </span>
            </div>
            <div
              class="cxd-StepsItem-description cxd-StepsItem-ellText"
            >
              <span />
            </div>
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
